components/script/dom/htmloptionelement.rs

```diff
@@ -93,6 +94,41 @@ impl HTMLOptionElement {
             select.ask_for_reset();
         }
     }
+
+    // https://html.spec.whatwg.org/multipage/#concept-option-index
+    fn index(&self) -> i32 {
+        if let Some(parent) = self.upcast::<Node>().GetParentNode() {
+            if let Some(select_parent) = parent.downcast::<HTMLSelectElement>() {
+                // return index in parent select's list of options
+                return self.index_in_select(select_parent);
+            } else if parent.downcast::<HTMLOptGroupElement>().is_some() {
```

Comment:
`parent.is::<HTMLOptGroupElement>()` is a bit easier to read.

Reply:
Oops, I think I just didn't know is::<T> existed, I will fix that.